### PR TITLE
core: fix some comments which not corresponding to code any more (#23…

### PR DESCRIPTION
**Pull request by SanJue**
…048)

### CODE REVIEW

**Comment by SanJue** [comment]
#23048

File: core/state/statedb.go [hunk 1]
```diff
@@ -832,8 +832,8 @@ func (s *StateDB) Finalise(deleteEmptyObjects bool) {
 }
 
 // IntermediateRoot computes the current root hash of the state trie.
-// It is called in between transactions to get the root hash that
-// goes into transaction receipts.
+// It is called in between blocks to get the root hash that
+// goes into block header.
```
Comment (karalabe) [inline core/state/statedb.go:836]: The comment here was correct. The method is used before the Byzantium fork blocks to return the hash after every tx.

File: core/state_transition.go [hunk 2]
```diff
@@ -43,7 +43,6 @@ The state transitioning model does all the necessary work to work out a valid ne
   4b) If valid, use result as code for the new state object
 == end ==
 5) Run Script section
-6) Derive new state root
```
Comment (karalabe) [inline core/state_transition.go:46]: It is still done for pre-Byzantium blocks, so the comment is correct.

File: core/state/statedb.go (hunk 1 above)
Comment (SanJue) [reply core/state/statedb.go:836]: I think it was correct before, but is wrong now.
Due to it's meaningful for old blocks, so I would not insist on it.
Thanks for your reply!

File: core/state_transition.go (hunk 2 above)
Comment (SanJue) [reply core/state_transition.go:46]: As above

**Comment by karalabe** [comment]
@holiman Pls confirm that these changes look good to you. You're more familiar with the weird quirks of the EVM :P

**Pull request closed without merging**